actions/cleanup-old-branches/scripts/cleanup-old-branches.sh

```diff
@@ -0,0 +1,216 @@
+declare -a OPEN_PR_BRANCHES=()
+
+if [[ -z "${GH_TOKEN:-}" ]]; then
+  echo "::error::GH_TOKEN environment variable is required. Exiting..."
+  exit 1
+fi
+
+if [[ -z "${GITHUB_REPOSITORY:-}" ]]; then
+  GITHUB_REPOSITORY=$(gh repo view --json 'nameWithOwner' --jq '.nameWithOwner')
+fi
+
+if [[ -z "${GITHUB_REPOSITORY_OWNER:-}" ]]; then
+  GITHUB_REPOSITORY_OWNER=$(echo "$GITHUB_REPOSITORY" | cut -d'/' -f1)
+fi
+
+GITHUB_REPOSITORY_NAME=$(echo "$GITHUB_REPOSITORY" | cut -d'/' -f2)
+
+# Fetch latest changes from origin.
+git fetch --prune "${GIT_REMOTE}"
+
+# Get the default base branch using the GitHub CLI.
+default_branch=$(gh repo view \
+  --json defaultBranchRef \
+  --jq .defaultBranchRef.name)
+
+echo "${section_delimiter}"
+echo "Configuration Details:"
+echo "${section_delimiter}"
+echo "Default branch is: $default_branch"
+echo "Branches newer than $DAYS_TO_KEEP days will be kept."
+echo "BRANCHES_TO_KEEP: $BRANCHES_TO_KEEP"
+echo "BRANCH_PREFIXES_TO_KEEP: $BRANCH_PREFIXES_TO_KEEP"
+echo "DRY_RUN: $DRY_RUN"
+echo "GIT_REMOTE: $GIT_REMOTE"
+echo "${section_delimiter}"
+echo
+
+# Check if the branch should be kept.
+# Returns: (zero for success, non-zero for failure)
+#  0 - Keep the branch.
+#  1 - Delete the branch.
+function should_keep_branch() {
+  local branch=$1
+  local days_old=$2
+  local branches_to_keep
+  local prefixes_to_keep
+
+  # Split comma-separated strings into arrays.
+  IFS=',' read -r -a branches_to_keep <<< "$BRANCHES_TO_KEEP"
+  IFS=',' read -r -a prefixes_to_keep <<< "$BRANCH_PREFIXES_TO_KEEP"
+
+  # Loop through exact branches to keep.
+  for kept_branch in "${branches_to_keep[@]}"; do
+    if [[ "$branch" == "$kept_branch" ]]; then
+      echo "$branch is in the list of branches to keep, preserving."
+      return 0
+    fi
+  done
+
+  # Loop through prefixes to keep.
+  for prefix in "${prefixes_to_keep[@]}"; do
+    if [[ "$branch" == "$prefix"* ]]; then
+      echo "$branch starts with a prefix to keep, preserving."
+      return 0
+    fi
+  done
+
+  # Loop through each open PR branch to check if the branch is associated with an open PR.
+  for open_branch in "${OPEN_PR_BRANCHES[@]}"; do
+    if [[ "$open_branch" == "$branch" ]]; then
+      echo "$branch is associated with an open PR, preserving."
+      return 0
+    fi
+  done
+
+  # If the branch is the default branch, keep it.
+  if [[ "$branch" == "$default_branch" ]]; then
+    echo "$branch is the default branch, preserving."
+    return 0
+  fi
+
+  # If the branch is newer than the threshold, keep it.
+  if [[ $days_old -le $DAYS_TO_KEEP ]]; then
+    echo "$branch is $days_old days old, preserving."
+    return 0
+  fi
+
+  return 1  # Delete this branch.
+}
+
+# Get list of head/feature branches that are associated with open pull requests.
+# XXX: Would have preferred to use this:
+#   gh pr list --state open --json headRefName --jq '.[].headRefName'
+# But it had a hard maximum of PR's and does not currently support pagination.
```

Comment:
hmm it has a `-L` limit flag no? like this?
`gh pr list -L 777777`

Reply:
it seems like it's `1000`:
https://github.com/cli/cli/blob/c9e8fd6c64f9030366990b5ce14ae2675226defd/pkg/cmd/attestation/api/client.go#L15